feat: add bootstrap step

rjerrems opened the pull request:
Adding the first part of the example foundation. I will submit each section as a separate PR to make it more digestible to review and document each individually.

ericyz commented on Makefile:
```diff
@@ -0,0 +1,43 @@
+docker_test_lint:
+	docker run --rm -it \
+		-v $(CURDIR):/workspace \
+		$(REGISTRY_URL)/${DOCKER_IMAGE_DEVELOPER_TOOLS}:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} \
```
How could I find what this image is about?

morgante commented on 0-bootstrap/README.md:
```diff
@@ -0,0 +1,57 @@
+## Usage
+
+1. Change into 0-bootstrap folder
+1. Rename terraform.example.tfvars to terraform.tfvars and update the file with values from your environment.
```
We should include explicit instructions (ex. `cp terraform.example.tfvars terraform.tfvars`.

rjerrems replied:
Make sense, fixed in next push.

morgante commented on 0-bootstrap/README.md:
```diff
@@ -0,0 +1,57 @@
+1. Run `terraform init`
+1. Run `terraform plan` and review output
+1. Run `terraform apply`
+1. Upload terraform state to created GCS bucket `gsutil cp terraform.tfstate gs://STATE_BUCKET/terraform/bootstrap/state/default.tfstate`
```
No need to do this manually. Once you reconfigure backend, Terraform prompts you to copy the state.

rjerrems replied:
This is cool, did not know about it. Have updated docs in next commit.

morgante commented on 0-bootstrap/backend.tf.example:
```diff
@@ -0,0 +1,6 @@
+    bucket = "STATE_BUCKET"
+    prefix = "terraform/bootstrap/state"
+  }
+}
```
Tiny nit: files should have a newline at the end.

morgante commented on Makefile:
```diff
@@ -0,0 +1,43 @@
+# Make will use bash instead of sh
+SHELL := /usr/bin/env bash
+
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.6.0
```
```suggestion
DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0
```

rjerrems commented:
Thanks for the review @morgante - I have fixed according to your feedback and added a PR for linting:

https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/609
